src/Entity/Policy.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace Drupal\password_enhancements\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+use Drupal\password_enhancements\Entity\Exception\PolicyException;
+
+/**
+ * Defines the Password Policy config entity.
+ *
+ * @ConfigEntityType(
+ *   id = "password_enhancements_policy",
+ *   label = @Translation("Password policy"),
+ *   label_collection = @Translation("Password policies"),
+ *   handlers = {
+ *     "storage" = "Drupal\password_enhancements\Entity\Storage\PolicyEntityStorage",
+ *     "route_provider" = {
+ *       "html" = "Drupal\password_enhancements\Entity\Routing\PolicyHtmlRouteProvider",
+ *     },
+ *     "form" = {
+ *       "default" = "Drupal\password_enhancements\Form\PolicyForm",
+ *       "delete" = "Drupal\password_enhancements\Form\PolicyDeleteForm",
+ *     },
+ *     "list_builder" = "Drupal\password_enhancements\Entity\PolicyListBuilder",
+ *   },
+ *   config_prefix = "policy",
+ *   admin_permission = "administer user password enhancements settings",
+ *   entity_keys = {
+ *     "id" = "role",
+ *     "label" = "role",
+ *   },
+ *   config_export = {
+ *     "expireSeconds",
+ *     "expireWarnSeconds",
+ *     "expiryWarningMessage",
+ *     "minimumRequiredConstraints",
+ *     "status",
+ *     "priority",
+ *     "role",
+ *   },
+ *   links = {
+ *     "collection" = "/admin/config/people/password-enhancements/policies",
+ *     "add-form" = "/admin/config/people/password-enhancements/policy/add",
+ *     "edit-form" = "/admin/config/people/password-enhancements/policy/{password_enhancements_policy}",
+ *     "delete-form" = "/admin/config/people/password-enhancements/policy/{password_enhancements_policy}/delete",
+ *   }
+ * )
+ */
+class Policy extends ConfigEntityBase implements PolicyInterface {
+
+  /**
+   * Expiry of the password in seconds, 0 means that passwords doesn't expire.
+   *
+   * @var int
+   */
+  protected $expireSeconds;
+
+  /**
+   * Show warning message before the password would expire in a given seconds.
+   *
+   * @var int
+   */
+  protected $expireWarnSeconds;
+
+  /**
+   * Expiry warning message.
+   *
+   * @var string|null
+   */
+  protected $expiryWarningMessage;
+
+  /**
+   * The number of constraints that are required.
+   *
+   * @var int
+   */
+  protected $minimumRequiredConstraints;
+
+  /**
+   * The priority of the policy.
+   *
+   * @var int
+   */
+  protected $priority;
+
+  /**
+   * Role ID.
+   *
+   * @var string
+   */
+  protected $role;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct(array $values, $entity_type) {
+    // Set initial values.
+    $this->expireSeconds = static::PASSWORD_NO_EXPIRY;
```

Comment:
All these could be initialized together with the property definitions :)

Reply:
Class declares properties, constructor define them.

Reply:
s/define/initialize/g  ;)
but it could happen outside of the constructor. Let's agree that we disagree. ;)